AirCasting/Utils/Bluetooth/BluetoothCommunicator.swift

```diff
@@ -0,0 +1,37 @@
+// Created by Lunar on 26/10/2022.
+//
+
+import Foundation
+
+struct CharacteristicUUID: Hashable {
+    let value: String
+}
+
+protocol BluetoothCommunicator {
+    typealias CharacteristicObserverAction = (Result<Data?, Error>) -> Void
+    
+    /// Adds an entry to observers of a particular characteristic
```

Comment:
other core protocols could also use this kind of documentation. It not only helps the user, but often helps to clear the vision while developing